test/unittest/tests/block/test_block_session.cpp

```diff
@@ -0,0 +1,286 @@
+}
+void
+validate_and_free_ext_block(WT_EXT *ext)
+{
+    validate_ext_block(ext);
+    free_ext_block(ext);
+}
+
+void
+free_ext_list(WT_BLOCK_MGR_SESSION *bms)
+{
+    WT_EXT *curr = bms->ext_cache;
+    while (curr != nullptr) {
+        WT_EXT *tmp = curr;
+        curr = curr->next[0];
+        __wt_free(nullptr, tmp);
+    }
+}
+
+void
+validate_and_free_ext_list(WT_BLOCK_MGR_SESSION *bms, int expected_items)
+{
+    validate_ext_list(bms, expected_items);
+    free_ext_list(bms);
+}
+
+TEST_CASE("Block session: __block_ext_alloc", "[block_session]")
+{
+    std::shared_ptr<MockSession> session = MockSession::buildTestMockSession();
+    WT_EXT *ext = nullptr;
+
+    __wt_random_init(&session->getWtSessionImpl()->rnd);
+
+    REQUIRE(__ut_block_ext_alloc(session->getWtSessionImpl(), &ext) == 0);
+    validate_and_free_ext_block(ext);
+}
+
+TEST_CASE("Block session: __block_ext_prealloc", "[block_session]")
+{
+    std::shared_ptr<MockSession> session = MockSession::buildTestMockSession();
+    WT_BLOCK_MGR_SESSION *bms = session->setupBlockManagerSession();
+
+    SECTION("Allocate zero extent blocks")
+    {
+        REQUIRE(__ut_block_ext_prealloc(session->getWtSessionImpl(), 0) == 0);
+        validate_and_free_ext_list(bms, 0);
+    }
+
+    SECTION("Allocate one extent block")
+    {
+        REQUIRE(__ut_block_ext_prealloc(session->getWtSessionImpl(), 1) == 0);
+        validate_and_free_ext_list(bms, 1);
+    }
+
+    SECTION("Allocate multiple extent blocks")
+    {
+        REQUIRE(__ut_block_ext_prealloc(session->getWtSessionImpl(), 3) == 0);
+        validate_and_free_ext_list(bms, 3);
+    }
+
+    SECTION("Allocate blocks on existing cache")
+    {
+        REQUIRE(__ut_block_ext_prealloc(session->getWtSessionImpl(), 3) == 0);
+        validate_ext_list(bms, 3);
+
+        REQUIRE(__ut_block_ext_prealloc(session->getWtSessionImpl(), 0) == 0);
+        validate_ext_list(bms, 3);
+
+        REQUIRE(__ut_block_ext_prealloc(session->getWtSessionImpl(), 2) == 0);
+        validate_ext_list(bms, 3);
+
+        REQUIRE(__ut_block_ext_prealloc(session->getWtSessionImpl(), 5) == 0);
+        validate_and_free_ext_list(bms, 5);
+    }
+}
+
+TEST_CASE("Block session: __wti_block_ext_alloc", "[block_session]")
+{
+    std::shared_ptr<MockSession> session = MockSession::buildTestMockSession();
+
+    SECTION("Allocate with null block manager session and no extent cache")
+    {
+        std::shared_ptr<MockSession> session_test_bm = MockSession::buildTestMockSession();
+        WT_EXT *ext;
+
+        REQUIRE(__wti_block_ext_alloc(session_test_bm->getWtSessionImpl(), &ext) == 0);
+        validate_and_free_ext_block(ext);
+
+        REQUIRE(__wti_block_ext_alloc(session->getWtSessionImpl(), &ext) == 0);
+        validate_and_free_ext_block(ext);
+    }
+
+    WT_BLOCK_MGR_SESSION *bms = session->setupBlockManagerSession();
+    SECTION("Allocate with fake zero cache extent count")
+    {
+        WT_EXT *ext;
+        WT_EXT *cached_ext;
```

Comment:
`cached_ext` can be declared just before it is used.

Reply:
[5f71e2e](https://github.com/wiredtiger/wiredtiger/pull/10963/commits/5f71e2efe324cb7f0f225d03d07a9041839c75f4)